File: jason/src/java/FIPAMessage.java
import com.google.gson.Gson;

class FIPAMessage {
    private final String performative;
    private final String sender;
    private final String receiver;
    private String content;

    public String getPerformative() {
        return performative;
    }

    public String getSender() {
        return sender;
    }

    public String getReceiver() {
        return receiver;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getContent() {
        return content;
    }

    public FIPAMessage(String performative, String sender, String receiver) {
        this.performative = performative;
        this.sender = sender;
        this.receiver = receiver;
    }

    public String encode() {
        Gson gson = new Gson();
        return gson.toJson(this);
    }

    public static FIPAMessage decode(String messageStr) {
        Gson gson = new Gson();
        return gson.fromJson(messageStr, FIPAMessage.class);
    }

    @Override
    public String toString() {
        return "FIPAMessage{" +
                "performative='" + performative + '\'' +
                ", sender='" + sender + '\'' +
                ", receiver='" + receiver + '\'' +
                ", content='" + content + '\'' +
                '}';
    }
}

File: front/src/app/models/FIPAMessage.ts
export class FIPAMessage {
  performative: string;
  sender: string;
  receiver: string;
  content: any;

  constructor(performative: string, sender: string, receiver: string, content: any) {
    this.performative = performative;
    this.sender = sender;
    this.receiver = receiver;
    this.content = content;
  }

  static decode(messageStr: string): FIPAMessage {
    const data = JSON.parse(messageStr);
    return new FIPAMessage(data.performative, data.sender, data.receiver, data.content);
  }
}